pro/src/ui-kit/form/PriceInput/PriceInput.tsx

```diff
@@ -0,0 +1,123 @@
+import { useField, useFormikContext } from 'formik'
+import { useState, useEffect, useRef } from 'react'
+
+import { BaseCheckbox } from 'ui-kit/form/shared/BaseCheckbox/BaseCheckbox'
+import { TextInput, TextInputProps } from 'ui-kit/form/TextInput/TextInput'
+
+type Price = number | ''
+
+/**
+ * Props for the PriceInput component.
+ *
+ * @extends Pick<TextInputProps, 'name' | 'max' | 'rightIcon' | 'disabled' | 'smallLabel' | 'className'>
+ */
+export type PriceInputProps = Pick<
+  TextInputProps,
+  | 'name'
+  | 'max'
+  | 'rightIcon'
+  | 'disabled'
+  | 'smallLabel'
+  | 'className'
+> & {
+  /**
+   * A label for the input,
+   * also used as the aria-label for the group.
+   */
+  label: string
+  /**
+   * A flag to show the "Gratuit" checkbox.
+   */
+  showFreeCheckbox?: boolean
+}
+
+/**
+ * The PriceInput component is a combination of a TextInput and a Checkbox to define prices.
+ * It integrates with Formik for form state management and is used when an null price is meant to be interpreted as free.
+ *
+ * ---
+ *
+ * @param {PriceInputProps} props - The props for the PriceInput component.
+ * @returns {JSX.Element} The rendered PriceInput component.
+ *
+ * @example
+ * <PriceInput
+ *  label="Price"
+ *  name="price"
+ *  max={100}
+ *  rightIcon={strokeEuroIcon}
+ * />
+ *
+ * @accessibility
+ * - **Labels**: Always provide a meaningful label using the `label` prop.
+ */
+export const PriceInput = ({
+  className,
+  name,
+  label,
+  max,
+  rightIcon,
+  disabled,
+  smallLabel,
+  showFreeCheckbox,
+}: PriceInputProps): JSX.Element => {
+  const priceName = name
```

Comment:
La variable est vraiment utile ?
```suggestion
export const PriceInput = ({
  className,
  priceName,
  label,
  max,
  rightIcon,
  disabled,
  smallLabel,
  showFreeCheckbox,
}: PriceInputProps): JSX.Element => {
```

Reply:
C'est purement pour la lisibilité du code.